dids/src/main/kotlin/web5/sdk/dids/DidDht.kt

```diff
@@ -0,0 +1,424 @@
+        requireNotNull(service.type) { "Service type cannot be null" }
+        requireNotNull(service.serviceEndpoint) { "Service serviceEndpoint cannot be null" }
+      }
+      // map to the DID object model's services
+      val services = opts.servicesToAdd?.map { service ->
+        Service.builder()
+          .id(URI.create("$id#${service.id}"))
+          .type(service.type)
+          .serviceEndpoint(service.serviceEndpoint)
+          .build()
+      }
+
+      // build DID Document
+      val didDocument =
+        DIDDocument.builder()
+          .id(URI(id))
+          .verificationMethods(verificationMethods)
+          .services(services)
+          .assertionMethodVerificationMethods(relationshipsMap[PublicKeyPurpose.ASSERTION_METHOD])
+          .authenticationVerificationMethods(relationshipsMap[PublicKeyPurpose.AUTHENTICATION])
+          .keyAgreementVerificationMethods(relationshipsMap[PublicKeyPurpose.KEY_AGREEMENT])
+          .capabilityDelegationVerificationMethods(relationshipsMap[PublicKeyPurpose.CAPABILITY_DELEGATION])
+          .capabilityInvocationVerificationMethods(relationshipsMap[PublicKeyPurpose.CAPABILITY_INVOCATION])
+          .build()
+
+      return DidDht(id, keyManager, didDocument)
+    }
+
+    /**
+     * Resolves a "did:dht" DID into a [DidResolutionResult], which contains the DID Document and possible related
+     * metadata.
+     *
+     * This implementation talks to a DID DHT gateway to retrieve the DID Document, which in turn takes the z-base-32
+     * encoded identifier public key and uses it to retrieve the DID Document from the DHT. Next, the Pkarr response is
+     * parsed, and used to reconstruct the DID Document.
+     *
+     * @param did The "did:dht" DID that needs to be resolved.
+     * @return A [DidResolutionResult] instance containing the DID Document and related context.
+     *
+     * @throws IllegalArgumentException if the provided DID does not conform to the "did:dht" method.
+     */
+    override fun resolve(did: String, options: ResolveDidOptions?): DidResolutionResult {
+      TODO("Not yet implemented")
+    }
+
+    /**
+     * Generates the identifier for a did:dht DID given its identity key.
+     *
+     * @param identityKey the key used to generate the DID's identifier
+     */
+    public fun getDidIdentifier(identityKey: JWK): String {
+      val publicKeyJwk = identityKey.toPublicJWK()
+      val publicKeyBytes = Crypto.publicKeyToBytes(publicKeyJwk)
+      val zBase32Encoded = ZBase32.encode(publicKeyBytes)
+      return "did:dht:$zBase32Encoded"
+    }
+
+    /**
+     * Checks whether a given DID identifier conforms to the "did:dht" method. This checks that the DID starts with
+     * "did:dht" and that the suffix is a valid z-base-32 encoded public key.
+     *
+     * @param did The DID to check.
+     * @return `true` if the DID conforms to the "did:dht" method, `false` otherwise.
+     */
+    public fun isValid(did: String): Boolean {
```

Comment:
added alternative suggestion for this function [in this PR](https://github.com/TBD54566975/web5-kt/pull/115)